In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import os

print("Notebook is working")

Notebook is working


**Data source:** https://www.kaggle.com/datasets/bytadit/ecommerce-order-dataset/data

# **1. Introduction**
I have 2 purposes for this project. To further practice data analysis with python and to do analysis on an ecommerce and supply chain dataset. 

The purpose of this project is to epxlore an ecommerce and supply chain dataset and find areas to analyze. Below is a list of the sections in this notebook:

1. Data inspection
2. Finding areas of inspection
3. Data processing and analysis
4. Conclusion

---

# **2. Data Inspection**
Before doing anything else, let's check the datasets that we have in the `data` directory.

In [3]:
# Current working directory
os.listdir("data")
os.getcwd()

'd:\\2. Areas\\Career_Exploration\\Data_Analyst\\ecommerce_sc_dataset'

Our dataset seems to be intended for machine learning as it is split into training and test data. For this specific project we'll focus mainly on EDA, wo we will not be modifying the data in any way. Thus it is safe to combine them both.

## 2.1. Concatenating datasets
First, let's read the training data into python.

In [4]:
# Get into 'train' working directory
os.chdir("d:\\2. Areas\\Career_Exploration\\Data_Analyst\\ecommerce_sc_dataset\\data\\train")
os.listdir(".")

# Reading the files into python
df_tr_cust = pd.read_csv("train_Customers.csv")
df_tr_ordItems = pd.read_csv("train_OrderItems.csv")
df_tr_orders = pd.read_csv("train_Orders.csv")
df_tr_payments = pd.read_csv("train_Payments.csv")
df_tr_products = pd.read_csv("train_Products.csv")

Now the testing data

In [5]:
# Get into 'test' working directory
os.chdir("d:\\2. Areas\\Career_Exploration\\Data_Analyst\\ecommerce_sc_dataset\\data\\test")
os.listdir(".")

# Reading the files into python
df_ts_cust = pd.read_csv("test_customers.csv")
df_ts_ordItems = pd.read_csv("test_OrderItems.csv")
df_ts_orders = pd.read_csv("test_Orders.csv")
df_ts_payments = pd.read_csv("test_Payments.csv")
df_ts_products = pd.read_csv("test_Products.csv")

Now that we have the data in python, we will now create copies and work with them instead of the raw data.

In [6]:
# Create copies
df_cust = pd.concat([df_ts_cust.copy(),df_tr_cust.copy()], ignore_index=True)
df_ordItems = pd.concat([df_ts_ordItems.copy(),df_tr_ordItems.copy()], ignore_index=True)
df_orders = pd.concat([df_ts_orders.copy(),df_tr_orders.copy()], ignore_index=True)
df_payments = pd.concat([df_ts_payments.copy(),df_tr_payments.copy()], ignore_index=True)
df_products = pd.concat([df_ts_products.copy(),df_tr_products.copy()], ignore_index=True)

In [26]:
df_all = pd.concat([df_cust, df_ordItems, df_orders, df_payments, df_products], ignore_index=True)
df_all.to_csv('temp_all_data.csv')

## 2.2. Inspection
Now that we have all the datasets combined, let's inspect each one.

#### Customer data

In [8]:
df_cust.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 127595 entries, 0 to 127594
Data columns (total 4 columns):
 #   Column                    Non-Null Count   Dtype 
---  ------                    --------------   ----- 
 0   customer_id               127595 non-null  object
 1   customer_zip_code_prefix  127595 non-null  int64 
 2   customer_city             127595 non-null  object
 3   customer_state            127595 non-null  object
dtypes: int64(1), object(3)
memory usage: 3.9+ MB


The combined customer has:

- 4 columns and 127,594 row
- no missing values
- correct data types

#### Order Items data

In [14]:
df_ordItems.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 127595 entries, 0 to 127594
Data columns (total 5 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   order_id          127595 non-null  object 
 1   product_id        127595 non-null  object 
 2   seller_id         127595 non-null  object 
 3   price             127595 non-null  float64
 4   shipping_charges  127595 non-null  float64
dtypes: float64(2), object(3)
memory usage: 4.9+ MB


This combined order items data has:

- 5 columns and 125,594 rows
- no missing values
- correct data types

#### Order data

In [17]:
df_orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 127595 entries, 0 to 127594
Data columns (total 7 columns):
 #   Column                         Non-Null Count   Dtype 
---  ------                         --------------   ----- 
 0   order_id                       127595 non-null  object
 1   customer_id                    127595 non-null  object
 2   order_purchase_timestamp       127595 non-null  object
 3   order_approved_at              127579 non-null  object
 4   order_status                   89316 non-null   object
 5   order_delivered_timestamp      87427 non-null   object
 6   order_estimated_delivery_date  89316 non-null   object
dtypes: object(7)
memory usage: 6.8+ MB


In [20]:
df_orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 127595 entries, 0 to 127594
Data columns (total 7 columns):
 #   Column                         Non-Null Count   Dtype 
---  ------                         --------------   ----- 
 0   order_id                       127595 non-null  object
 1   customer_id                    127595 non-null  object
 2   order_purchase_timestamp       127595 non-null  object
 3   order_approved_at              127579 non-null  object
 4   order_status                   89316 non-null   object
 5   order_delivered_timestamp      87427 non-null   object
 6   order_estimated_delivery_date  89316 non-null   object
dtypes: object(7)
memory usage: 6.8+ MB


This combined order data has:

- 7 columns and 127,594 rows
- missing values in columns `order_approved_at`, `order_status`, `order_delivered_timestamp`, and `order_estimated_delivery_date`.
- correct data types, however dates and timestamps will need to be covnerted into their proper formats in the processing.

#### Payments data

In [18]:
df_payments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 127595 entries, 0 to 127594
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              127595 non-null  object 
 1   payment_sequential    127595 non-null  int64  
 2   payment_type          127595 non-null  object 
 3   payment_installments  127595 non-null  int64  
 4   payment_value         127595 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.9+ MB


This combined payments data has:

- 5 columns and 127,594 rows
- no missing values
- correct data types

#### Products data

In [19]:
df_products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 127595 entries, 0 to 127594
Data columns (total 6 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   product_id             127595 non-null  object 
 1   product_category_name  127119 non-null  object 
 2   product_weight_g       127570 non-null  float64
 3   product_length_cm      127570 non-null  float64
 4   product_height_cm      127570 non-null  float64
 5   product_width_cm       127570 non-null  float64
dtypes: float64(4), object(2)
memory usage: 5.8+ MB


This combined products data has:

- 6 columns and 127,594 rows
- missing values observed across all columns
- correct data types

### Overall dataset structures
This dataset seems to be structurally sound, with the only issue being missing values in some columns.

---


# **3. Areas of Analysis**
Since there are 23 columns in total, there's quite a few areas of analysis that we can choose.

##### Lead Time

Lead time is one of the many aspects of a supply chain that reflects how efficient a company is. In this analysis we'll be able to see how efficient this company is at handling orders to delivery. 

##### On-Time Delivery
On-time delivery is another important metric in the supply chain. We will be comparing how many recorded shipments were on time, early, or late. 

##### Ordered Items Category per Area
The number of each product category being ordered per area could imply real needs that the company could focus on providing more

However, this analysis will focus on top 5 most frequently purchased items only to avoid excessive visual complexity.

##### Payment Type Preference and Installments per Area
The type of payment chosen by customers and the number of payment installments in each area could potentially show the financial states and spending habits of the customers in the area. 

However, this analysis will be done at an aggregate level, with regional comparisons limited to the highest-volume states to maintain interpretability.